### ISSUE TITLE
mixed type routes issue?

### ISSUE THREAD

**Issue by mbouclas** [angular/router#188]
So i wanted to try something very basic, an app with two routes, one of which is consisted of two components.

The route config is : 
```
        $router.config([
            {
                path: '/',
                component: {
                    navigation: 'navigation',
                    main: 'welcome'
                }
            },
            {
                path: '/user/:userId',
                component: 'user'
            }
        ]);
```

Here is my problem, if i use
```
<ng-viewport></ng-viewport>
```
in my HTML the app crashes cause it's missing the explicit declarations in the HTML. But, if i do this : 
```
<nav ng-viewport="navigation"></nav>
<main ng-viewport="main"></main>
```

Then, when navigating to the user route, i get the user component shown twice, once in each viewport. So how do we go about this ? 

On a side note, i think that nesting the routes would be a good idea (even if nesting up to a certain level) cause in the case above, if you want to re-use the navigation component, you have to add it to each possible route.

**Comment by btford** [comment]
Can you please provide a reproduction via [jsbin](http://jsbin.com/), [plnkr.co](http://plnkr.co), or [jsfiddle](http://jsfiddle.net/)?

This saves us a lot of time. Thanks!

**Comment by mbouclas** [comment]
Yes, check this https://github.com/mbouclas/mcms-node-new-ngroute-test
i have running instructions on the README

**Comment by nicksloan** [comment]
@mbouclas A Github repo with installation instructions creates a burden for the people triaging and fixing your issue, which means it is less likely to be fixed. You should try to use one of the requested services next time. Fortunately, your example is also workable from RawGit.com:

@btford Here's a working URL: https://rawgit.com/mbouclas/mcms-node-new-ngroute-test/master/public/index.html
